src/strtree.c

```diff
@@ -0,0 +1,175 @@
+#define PY_SSIZE_T_CLEAN
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+
+#include <Python.h>
+#include <structmember.h>
+
+#define NO_IMPORT_ARRAY
+#define PY_ARRAY_UNIQUE_SYMBOL pygeos_ARRAY_API
+
+#include <numpy/arrayobject.h>
+#include <numpy/ndarraytypes.h>
+#include <numpy/npy_3kcompat.h>
+
+#include "strtree.h"
+#include "geos.h"
+#include "pygeom.h"
+#include "kvec.h"
+
+
+static PyObject *STRtree_new(PyTypeObject *type, PyObject *args,
+                             PyObject *kwds)
+{
+    int node_capacity;
+    PyObject *arr;
+    void *tree, *ptr;
+    npy_intp n, i;
+    GEOSGeometry *geom;
+    GeometryObject *obj;
+    GEOSContextHandle_t context = geos_context[0];
+
+    if (!PyArg_ParseTuple(args, "Oi", &arr, &node_capacity)) {
+        return NULL;
+    }
+    if (!PyArray_Check(arr)) {
+        PyErr_SetString(PyExc_TypeError, "Not an ndarray");
+        return NULL;
+    }
+    if (!PyArray_ISOBJECT((PyArrayObject *) arr)) {
+        PyErr_SetString(PyExc_TypeError, "Array should be of object dtype");
+        return NULL;
+    }
+    if (PyArray_NDIM((PyArrayObject *) arr) != 1) {
+        PyErr_SetString(PyExc_TypeError, "Array should be one dimensional");
+        return NULL;
+    }
+
+    tree = GEOSSTRtree_create_r(context, (size_t) node_capacity);
+    if (tree == NULL) {
+        return NULL;
+    }
+
+    n = PyArray_SIZE((PyArrayObject *) arr);
+    for(i = 0; i < n; i++) {
+        /* get the geometry */
+        ptr = PyArray_GETPTR1((PyArrayObject *) arr, i);
+        obj = *(GeometryObject **) ptr;
+        /* skip incase obj was no geometry or None */
+        if (!get_geom(obj, &geom)) { continue; }
```

Comment:
Shouldn't we rather raise an error in case of not geometry/None? (that's what we do for all ufuncs I think?)

Reply:
Agreed. Not geometry -> error ; None -> empty array